wdl-analysis/src/document/v1.rs

```diff
@@ -1169,117 +1171,106 @@ fn add_call_statement(
-                    .inputs
-                    .get(input_name.as_str())
-                    .copied()
-                    .map(|i| i.ty)
-                    .unwrap_or_else(|| {
-                        diagnostics.push(unknown_io_name(
-                            name.as_str(),
-                            &input_name,
-                            target.workflow,
-                            true,
-                        ));
-                        Type::Union
-                    });
-
-                match input.expr() {
-                    Some(expr) => {
-                        type_check_expr(
-                            document,
-                            ScopeRef::new(scopes, index),
-                            &expr,
-                            expected_ty,
-                            input_name.span(),
-                            diagnostics,
-                        );
-                    }
-                    None => {
-                        if let Some(name) = ScopeRef::new(scopes, index).lookup(input_name.as_str())
+            match input.expr() {
+                Some(expr) => {
+                    type_check_expr(
+                        document,
+                        ScopeRef::new(scopes, index),
+                        &expr,
+                        expected_ty,
+                        input_name.span(),
+                        diagnostics,
+                    );
+                }
+                None => {
+                    if let Some(name) = ScopeRef::new(scopes, index).lookup(input_name.as_str()) {
+                        if expected_ty != Type::Union
+                            && !name.ty.is_coercible_to(&document.types, &expected_ty)
                         {
-                            if expected_ty != Type::Union
-                                && !name.ty.is_coercible_to(&document.types, &expected_ty)
-                            {
-                                diagnostics.push(call_input_type_mismatch(
-                                    &document.types,
-                                    &input_name,
-                                    expected_ty,
-                                    name.ty,
-                                ));
-                            }
+                            diagnostics.push(call_input_type_mismatch(
+                                &document.types,
+                                &input_name,
+                                expected_ty,
+                                name.ty,
+                            ));
                         }
                     }
                 }
-
-                // Don't bother keeping track of seen inputs if nested inputs are allowed
-                if !nested_inputs_allowed {
-                    seen.insert(TokenStrHash::new(input_name));
-                }
             }
 
+            // Don't bother keeping track of seen inputs if nested inputs are allowed
```

Comment:
We could emit a warning.

The work I'm about to put up validates input JSONs against a document and one of the thing it turns into an error is a missing nested input when `allow_nested_inputs` is `true`.

However, we'd want to suppress the warning for he analysis performed by the engine as any missing call inputs that aren't present in the inputs given the engine will cause an error.

Reply:
Can definitely think of some uses (specifically in our CI) for that input JSON validation. That would be good for a PR I have in review, so excited to test that out. But I would still want this `warning` emitted during `sprocket check`.

Can you elaborate on your last paragraph? I don't think I'm following.

> However, we'd want to suppress the warning for he analysis performed by the engine as any missing call inputs that aren't present in the inputs given the engine will cause an error.

Reply:
The engine performs an analysis before it evaluates any WDL; so we don't want such a "missing input" warning emitted in that case because the input might be coming in from JSON (or programmatically supplied).

Reply:
If we do emit a warning, I think it'd be for a future PR.

Reply:
Ah gotchya. That makes sense. Agreed it should wait for another PR.